### PR TITLE
Fixed the shared object issue in the controller task return.

### PR DESCRIPTION
**Pull request by yhwen**
Fixes # .

Fixed the simulator delay with large number of clients run.

### Description

A shared memory object issue in the controller task return caused the concurrent clients get_task request got the same task to run. This caused the simulator clients ran for multiple times.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/apis/impl/controller.py [hunk 1]
```diff
@@ -287,7 +287,7 @@ def process_task_request(self, client: Client, fl_ctx: FLContext) -> Tuple[str,
                 task.last_client_task_map[client.name] = client_task_to_send
                 task.client_tasks.append(client_task_to_send)
                 self._client_task_map[client_task_to_send.id] = client_task_to_send
-            return task_name, client_task_to_send.id, task_data
+            return task_name, client_task_to_send.id, copy.deepcopy(task_data)
```
Comment (IsaacYangSLA) [inline nvflare/apis/impl/controller.py:290]: This may be using lots of memory as task_data is a shareable.

Comment (yhwen) [reply nvflare/apis/impl/controller.py:290]: I am aware of this. Ideally, we need only the copy the headers part of the shareable. But we don't that enabled in the Sharable. We can enhance that in the future.

Comment (YuanTingHsieh) [reply nvflare/apis/impl/controller.py:290]: The upstream (who called this method) is trying to set header to this task_data's shareable.
Resulting in multiple threads trying to write/set same key to different values at the same time.

By returning a full copy, we don't write/set the same object.

File: nvflare/private/fed/server/server_commands.py [hunk 2]
```diff
@@ -166,7 +164,10 @@ def process(self, data: Shareable, fl_ctx: FLContext):
         shared_fl_ctx.set_public_props(copy.deepcopy(get_serializable_data(fl_ctx).get_all_public_props()))
         shareable.set_header(key=FLContextKey.PEER_CONTEXT, value=shared_fl_ctx)
 
-        # return fobs.dumps(data)
+        self.logger.info(
+            f"return task to client.  client_name:{client.name}  task_id:{task_id}  "
+            f"sharable_header_task_id: {shareable.get_header(key=FLContextKey.TASK_ID)}"
+        )
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/server/server_commands.py:170]: do we want this log at info level anymore?

File: nvflare/private/fed/server/server_commands.py [hunk 3]
```diff
@@ -209,6 +203,7 @@ def process(self, data: Shareable, fl_ctx: FLContext):
         task_id = data.get_cookie(FLContextKey.TASK_ID)
         server_runner = fl_ctx.get_prop(FLContextKey.RUNNER)
         server_runner.process_submission(client, contribution_task_name, task_id, data, fl_ctx)
+        self.logger.info(f"submit_update process. client_name:{client.name}   task_id:{task_id}")
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/server/server_commands.py:206]: do we want this log at info level anymore?

Comment (yhwen) [reply nvflare/private/fed/server/server_commands.py:206]: same as above.

File: nvflare/private/fed/server/server_commands.py (hunk 2 above)
Comment (YuanTingHsieh) [reply nvflare/private/fed/server/server_commands.py:170]: sounds good, I am OK to both situations.

**Pull request merged**